### DISCUSSION TITLE
Manage DNS

### DISCUSSION THREAD

**Discussion by roupiemaxime** [auroraresearchlab/netbox-dns#182] [General]
Hello everyone,

What type of DNS is supported ? Bind ?Windows ?
DNS registration is done automatically as on IPAM ?

Thanks a lot for your feedback,
Max.

**Comment by peteeckel** [comment]
Hi, 

the idea behind NetBox DNS is not to manage a DNS **server**, but to maintain DNS **data**. That is in line with the general idea of NetBox as a source of truth, i.e. something to have your data in. 

The upside of this is that you can use the data from NetBox DNS to provision any server, the downside is that you have to do it yourself. There are a couple of options to do this. 

1. You can write some code of whatever kind that plugs into the NetBox REST API and gets the data from there, and has an interface to whatever DNS server you like and populates it with the data.
2. You can use the Python `pynetbox` module for the connection to NetBox DNS, saving you the onus of writing the REST API code yourself. The backend is still your responsibliity though.
3. You can use Ansible and the `netbox` Ansible module to abstract a level higher (Ansible in turn uses pynetbox). If there is another Ansible module for your DNS server (there are several), you can spare yourself the effort of writing any non-Ansible code altogether.
4. There is also an [octodns-Provider](https://github.com/jcollie/octodns-netbox-dns) by @jcollie that you can use if your DNS server or service is supported by octodns. That should be even simpler (haven't tried it yet, though). You could, for instance, provision PowerDNS via octodns and use it as your master server.

In the documentation section there is a very simple example of how to create zone files for BIND from NetBox DNS data. Have a look at [docs/using_netbox_dns_with_ansible.md](https://github.com/auroraresearchlab/netbox-dns/blob/main/docs/using_netbox_dns_with_ansible.md). This is not very elaborate and currently does not support views yet, but it should be enough to get the general idea of how to get data out of NetBox DNS and use them.

> DNS registration is done automatically as on IPAM?

What do you mean by that? If the idea is that the data you enter into NetBox DNS will automatically be activated in your DNS server - no. Some reasons were discussed in #8, and although I found the idea interesting at first I am now convinced that it isn't such a good one after all. But you could write a Webhook that takes care of automatically get the data into the DNS server's dataset.

**Reply by robertlynch3** [reply]
> > DNS registration is done automatically as on IPAM?
> 
> What do you mean by that? If the idea is that the data you enter into NetBox DNS will automatically be activated in your DNS server - no. Some reasons were discussed in #8, and although I found the idea interesting at first I am now convinced that it isn't such a good one after all. But you could write a Webhook that takes care of automatically get the data into the DNS server's dataset.

I think @roupiemaxime is referring to, if you register an IP in IPAM and enter a DNS name, is there a way to have it then add said DNS name to the zone file, or do they need to be two separate operations with no foreign key relationship.

**Comment by peteeckel** [comment]
Have a look at PR #223, which adds example code for a custom script that can be used to export zone data. Depending on what you want to acheve that might or might not be useful - at least as a base for your own developments.
